### ISSUE TITLE
Problem with delimiters

### ISSUE THREAD

**Issue by heiderich** [lervag/vimtex#616]
### Explain the issue

Since one of the last commits (I am on 7b35d3782ba2284a108dc5ae3bd0b4cfe04a80c1 now) I experience problems when typing delimiters in math-mode.

#### Steps to reproduce

Open the a file with the following content:

```tex
\documentclass[12pt]{article}
\begin{document}
\begin{enumerate}
        \item
\end{enumerate}
\end{document}
```

Place the cursor behind \item then type `$(` in insert mode.
#### Expected behaviour

`$(` appears.

#### Observed behaviour

vim hangs using 100% of the CPU.

### Minimal working example

see above

### Minimal vimrc file

``` vim
set nocompatible

" Load Vimtex
let &rtp  = '~/.vim/bundle/vimtex,' . &rtp
let &rtp .= ',~/.vim/bundle/vimtex/after'

" Load other plugins, if necessary
" let &rtp = '~/path/to/other/plugin,' . &rtp

filetype plugin indent on
syntax enable

" Vimtex options go here
```

**Comment by yous** [comment]
Yes, recently I have to press `^C` to interrupt the plugin and input something else, or I have to type `$$` first, and then move the cursor between two `$` symbols to type the content of math equations.

**Comment by John-Colvin** [comment]
I get this as well, makes the plugin effectively unusable for me.

**Comment by lervag** [comment]
Sorry about that. I think it is fixed now, please test.

**Comment by John-Colvin** [comment]
Thanks, it doesn't hang for anywhere near as long now but there's still about a 0.5s delay for me every time I type any delimiter character e.g. `}`

**Comment by lervag** [comment]
Good to hear it's getting better!

In order for me to investigate and (perhaps) improve the efficiency further, could you please open a new issue, again with a proper minimal working example? That would be very helpful for me to investigate the mentioned delays. Note that I currently do not experience any such delays while writing myself. This could be because of faster hardware, but I think it might rather be because I hardwrap my lines (i.e. I only have short lines), or because I currently don't have that much math in my documents.

So, to summarize: Please open new issues with good MWEs to help me further investigate the delays.
